src/tests/cli_tests.py

```diff
@@ -673,219 +542,408 @@ def rnp_cleartext_signing_gpg_to_rnp(filesize):
+            rnp_sym_encryption_rnp_to_gpg(cipher, size, 6, 'zip')
+            rnp_sym_encryption_rnp_to_gpg(cipher, size, 6, 'zlib')
+            rnp_sym_encryption_rnp_to_gpg(cipher, size, 6, 'bzip2')
+
+class Compression(unittest.TestCase):
+
+    def setUp(self):
+        # Compression is currently implemented only for encrypted messages
+        rnp_genkey_rsa(KEY_ENCRYPT)
+        rnp_genkey_rsa(KEY_SIGN_GPG)
+        gpg_import_pubring()
+        gpg_import_secring()
+
+    def tearDown(self):
+        clear_keyrings()
 
-    # Parameters are ok so we can proceed
+    def test_rnp_compression(self):
+        levels = [0, 2, 4, 6, 9]
+        algosrnp = ['zip', 'zlib', 'bzip2']
+        algosgpg = [1, 2, 3]
+        sizes = [20, 1000, 5000, 20000, 150000, 1000000]
 
-    setup()
+        for size in sizes:
+            for algo in [0, 1, 2]:
+                for level in levels:
+                    gpg_to_rnp_encryption(
+                        'AES', size, level, algosgpg[algo])
+                    file_encryption_rnp_to_gpg(size, level, algosrnp[algo])
+                    rnp_signing_gpg_to_rnp(size, level, algosgpg[algo])
+
+class Sign(unittest.TestCase):
+    '''
+        Things to try later:
+        - different public key algorithms
+        - different hash algorithms where applicable
+        - cleartext signing/verification
+        - detached signing/verification
+    '''
+    # Message sizes to be tested
+    SIZES = [20, 1000, 5000, 20000, 150000, 1000000]
+
+    @classmethod
+    def setUpClass(cls):
+        # Generate keypair in RNP
+        rnp_genkey_rsa(KEY_SIGN_RNP)
+        rnp_genkey_rsa(KEY_SIGN_GPG)
+        gpg_import_pubring()
+        gpg_import_secring()
+
+    @classmethod
+    def tearDownClass(cls):
+        clear_keyrings()
 
-    if 'rnpkeys' in tests:
-        run_rnpkeys_tests()
-    if 'rnp' in tests:
-        run_rnp_tests()
+    # TODO: This script should generate one test case per message size.
+    #       Not sure how to do it yet
+    def test_rnp_to_gpg_default_key(self):
+        for size in Sign.SIZES:
+            rnp_signing_rnp_to_gpg(size)
+            rnp_detached_signing_rnp_to_gpg(size)
+            rnp_cleartext_signing_rnp_to_gpg(size)
 
-    succeeded = len(TESTS_SUCCEEDED)
-    failed = len(TESTS_FAILED)
-    print '\nRun {} tests, {} succeeded and {} failed.\n'.format(succeeded + failed, succeeded, failed)
-    if failed > 0:
-        print 'Failed tests:\n' + '\n'.join(TESTS_FAILED)
-        sys.exit(1)
+    def test_gpg_to_rnp_default_key(self):
+        for size in Sign.SIZES:
+            rnp_signing_gpg_to_rnp(size)
+            rnp_detached_signing_gpg_to_rnp(size)
+            rnp_cleartext_signing_gpg_to_rnp(size)
 
-def cleanup():
-    if DEBUG:
-        return
+if __name__ == '__main__':
+    main = unittest.main
+    main.USAGE = "\nRNP test client specific flags:\n  -w,\t\t Don't remove working directory" + main.USAGE
```

Comment:
Looks like this needs another newline before `main.USAGE`.

Reply:
Not sure why it would need it.

Reply:
Right now it looks like this for me:
```
RNP test client specific flags:
  -w,		 Don't remove working directoryUsage: cli_tests.py [options] [test] [...]
```

With a couple newlines:
```
RNP test client specific flags:
  -w,		 Don't remove working directory

Usage: cli_tests.py [options] [test] [...]
```

Reply:
ah, I see what's you mean. Agreed,